core/src/test/java/com/radixdlt/genesis/olympia/converter/OlympiaToBabylonGenesisConverterTest.java

```diff
@@ -0,0 +1,602 @@
+            .collect(
+                Collectors.toMap(r -> HashCode.fromBytes(r.addressBytesWithoutEntityId()), r -> r));
+
+    final var babylonAllocationsByResource =
+        new HashMap<ResourceAddress, Map<ComponentAddress, Decimal>>();
+    resourceBalances.forEach(
+        resourcesChunk -> {
+          resourcesChunk
+              .allocations()
+              .forEach(
+                  t -> {
+                    final var resourceAddr = t.first();
+                    final var resourceAllocations =
+                        babylonAllocationsByResource.computeIfAbsent(
+                            resourceAddr, unused -> new HashMap<>());
+                    t.last()
+                        .forEach(
+                            alloc -> {
+                              final var acc =
+                                  resourcesChunk.accounts().get(alloc.accountIndex().toInt());
+                              resourceAllocations.put(acc, alloc.amount());
+                            });
+                  });
+        });
+
+    stateSummary.resources.forEach(
+        summaryResource -> {
+          final var babylonAddrBytes =
+              Bytes.leftPadWithZeros(
+                  summaryResource.resourceAddrBytes.asBytes(), ResourceAddress.BYTE_LENGTH - 1);
+          final var babylonResource =
+              babylonResourcesByAddr.get(HashCode.fromBytes(babylonAddrBytes));
+          assertEquals(
+              summaryResource.expectedScaledTotalSupply(), babylonResource.initialSupply());
+
+          final var babylonBalances =
+              babylonAllocationsByResource.get(
+                  toGlobalFungibleAddr(summaryResource.resourceAddrBytes.asBytes()));
+          summaryResource.balances.forEach(
+              (accKey, amount) -> {
+                final var babylonAmount =
+                    babylonBalances.get(Address.virtualAccountAddress(accKey.asBytes()));
+                assertEquals(
+                    scaleResourceAmount(
+                        amount,
+                        summaryResource.totalSupply(),
+                        CONVERTER_CONFIG
+                            .maxGenesisResourceUnscaledSupply()
+                            .toRawBigIntRepresentation()),
+                    babylonAmount);
+              });
+        });
+  }
+
+  private static Decimal scaleResourceAmount(
+      BigInteger originalAmount,
+      BigInteger resourceTotalSupplyOnOlympia,
+      BigInteger resourceMaxSupplyOnBabylon) {
+    if (resourceTotalSupplyOnOlympia.compareTo(resourceMaxSupplyOnBabylon) <= 0) {
+      return Decimal.unsafeFromRawBigIntRepr(originalAmount);
+    } else {
+      final var scaledBigInt =
+          resourceMaxSupplyOnBabylon.multiply(originalAmount).divide(resourceTotalSupplyOnOlympia);
+      return Decimal.unsafeFromRawBigIntRepr(scaledBigInt);
+    }
+  }
+
+  private <T> void verifyExpectedChunksCount(int numElements, int maxElements, List<T> chunks) {
```

Comment:
Can we also check that the total number of entries across all the chunks is the same as `maxElements`?